src/transformers/deepspeed.py

```diff
@@ -116,37 +110,67 @@ class HfTrainerDeepSpeedConfig(HfDeepSpeedConfig):
 
     def __init__(self, config_file_or_dict):
         super().__init__(config_file_or_dict)
+        self.mismatches = []
+
+    def auto(self, ds_key_long, hf_val, hf_key=None, must_match=False):
+        """
+        A utility method that massages the config file and can optionally verify that the values match.
+
+        1. Replace "auto" values with ``TrainingArguments`` value.
+
+        2. If it wasn't "auto" and ``must_match`` is true, then check that DS config matches Trainer
+        config values and if mismatched add the entry to ``self.mismatched`` - will assert during
+        ``trainer_config_finalize`` for one or more mismatches.
+
+        """
+
+        c = self.config
+
+        # find the config node of interest if it exists
+        nodes = ds_key_long.split(".")
+        ds_key = nodes.pop()
+        for node in nodes:
+            c = c.get(node)
+            if c is None:
+                return
+
+        if c.get(ds_key) == "auto":
+            c[ds_key] = hf_val
+            return
+
+        if not must_match:
+            return
+
+        ds_val = c.get(ds_key)
+        if ds_val is not None and ds_val != hf_val:
+            self.mismatches.append(f"- ds {ds_key_long}={ds_val} vs hf {hf_key}={hf_val}")
+
+    auto_match = partialmethod(auto, must_match=True)
```

Comment:
Could be `auto_fill_and_match` if `auto` is renamed, but `auto_match` works as well.

Reply:
Your suggestions are great, but I was just trying to keep these short to avoid the auto-formatter making the really readable code much less readable due to forced inconsistent wrapping. 

May be I can use `fill_only` and `fill_match`? or just damn formatting and use better function names.

Reply:
Works for me!